FEAT : ODYA-218 로그인 화면 hi-fi 적용 및 로그인 기능 구현

leeJongMin58 opened the pull request:
### 개요

- 로그인 화면 hi-fi 적용 및 로그인 기능 구현
- 스플레쉬 화면에서 로그인 여부 확인 후 메인 또는 로그인 창으로 이동
- 토픽 선택 부분 제외

### 변경사항

### 관련 지라 및 위키 링크
https://weit.atlassian.net/browse/ODYA-218?atlOrigin=eyJpIjoiZTUyYmNmZDBkZTI5NDJmMWJkOGI3ZDQ4NDJiOWYyZmYiLCJwIjoiaiJ9

### 리뷰어에게 하고 싶은 말
카카오 로그인 로고 관련해서 질문 있는데 슬랙으로 할게요

위에 영상은 회원가입이 되어 있지 않을때 / 뒤에 영상은 회원가입이 되어 있을때 입니다

https://github.com/weIT-1st/Odya-android/assets/151716813/699ad1e1-878d-4088-8591-2396019fde67


https://github.com/weIT-1st/Odya-android/assets/151716813/63fa80e9-d2a7-45cc-95df-c52d442d9bae

Commit:
CHORE : 로그인 화면 hi- fi 적용

- 로그인 화면 hi-fi 적용 및 로그인 기능 구현
- 스플레쉬 화면에서 로그인 여부 확인 후 메인 또는 로그인 창으로 이동
- 토픽 선택 부분 제외

minseonglove commented on presentation/src/main/java/com/weit/presentation/ui/login/input/nickname/LoginNicknameFragment.kt:
```diff
@@ -38,10 +37,31 @@ class LoginNicknameFragment : BaseFragment<FragmentLoginNicknameBinding>(
+    }
+
+    @SuppressLint("ClickableViewAccessibility")
+    private fun deleteEditTextAll(){
```
initListener에 그대로 정의해도 되겠어요. deleteEditTextAll이라는 메소드는 리스너를 등록한다기 보단 진짜로 텍스트를 다 날려버리는 느낌으로 보여요

overthename commented on presentation/src/main/java/com/weit/presentation/ui/login/input/nickname/LoginNicknameFragment.kt:
```diff
@@ -38,10 +37,31 @@ class LoginNicknameFragment : BaseFragment<FragmentLoginNicknameBinding>(
+        etLoginInputName.setOnTouchListener(
+            View.OnTouchListener { v, event ->
+                if (event.action == MotionEvent.ACTION_UP) {
+                    if (event.x >= (etLoginInputName.right - etLoginInputName.compoundDrawables[2].bounds.width())) {
```
여기 코드는 무슨 의미인가요?

leeJongMin58 replied:
엑스 버튼 눌렀을때 에딧텍스트에 글 전체 삭제요

overthename reviewed (approved):
수고하셨습니당

Commit:
FEAT : 수정 ver2

Commit:
FEAT : 수정 ver1

더 나은 서비스를 위해서 권한 동의가 필요해요 -> 스크롤 뷰

onboarding -> viewpager로 view holder 처럼

png  -> wepb / 메인페이지를 생성중이에요 => 텍스트뷰

로그인 닉네임  창 ==> 닉네임 설정 기준을 표시해 주어야 한다. 디자인팀에게 물어보기

spinner adapter -> position 말고 enum으로 바꿔주기

object -> 항상 같은 놈이 나오는지 확인하고 지역 변수로 설정해주기

회원가입 등록 ->  스낵바를 액티비에서 띄우기 + 로그인 성공시 스낵바 지우기

minseonglove commented on presentation/src/main/java/com/weit/presentation/ui/login/input/userinfo/LoginInputUserInfoFragment.kt:
```diff
@@ -42,6 +43,25 @@ class LoginInputUserInfoFragment : BaseFragment<FragmentLoginInputUserInfoBindin
+                viewModel.setGender(gender)
+            }
+
+            override fun onNothingSelected(parent: AdapterView<*>?) {
+                TODO("Not yet implemented")
+            }
```
이건 정의를 아직 하지 않았다는 의미에요.
아무 정의도 안할것이기 때문에 주석을 적절하게 바꿔주세요

minseonglove commented on presentation/src/main/java/com/weit/presentation/ui/login/onboarding/LoginOnboardingAdapter.kt:
```diff
@@ -0,0 +1,50 @@
+        val item = getItem(position)
+        holder.bind(item)
+    }
+    inner class ViewHolder(
+        private val binding: ItemLoginOnboardingBinding
+    ): RecyclerView.ViewHolder(binding.root){
```
inner일 필요는 없어뵈네용

minseonglove commented on presentation/src/main/java/com/weit/presentation/ui/login/onboarding/LoginOnboardingFragment.kt:
```diff
@@ -0,0 +1,84 @@
+) {
+    private val viewModel: LoginOnboardingViewModel by viewModels()
+    private val loginOnboardingAdapter = LoginOnboardingAdapter()
+    private val pageChangeCallback = object : ViewPager2.OnPageChangeCallback() {
+        override fun onPageSelected(position: Int) {
+            Log.d("jomi", "listener : $position")
+            viewModel.changeCurrentPage(position)
+        }
+    }
```
로그 지우기

minseonglove commented on presentation/src/main/java/com/weit/presentation/ui/login/onboarding/LoginOnboardingFragment.kt:
```diff
@@ -0,0 +1,84 @@
+        }
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        viewModel
+        initLoginOnboardingViewPager()
+    }
```
ㅋㅋ 모야 이 viewmodel

minseonglove reviewed (commented):
조습니다 수정하구 머지해주세용

Commit:
FEAT : 수정 ver2

Commit:
FEAT : 수정 ver2

overthename reviewed (approved):
굿

Commit:
Merge branch 'develop' into feature/ODYA-218

# Conflicts:
#	presentation/src/main/res/drawable/selectro_dot_indicator.xml
#	presentation/src/main/res/values/styles.xml
